src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaTests.cs

```diff
@@ -0,0 +1,83 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Security.Cryptography.SLHDsa.Tests
+{
+    [ConditionalClass(typeof(SlhDsaTestsBase), nameof(SupportedOnPlatform))]
+    public class SlhDsaTests : SlhDsaTestsBase
+    {
+        [ConditionalFact(nameof(SupportedOnPlatform))]
+        public static void NulArgumentValidation_Create()
+        {
+            Assert.Throws<ArgumentNullException>(() => SlhDsa.GenerateKey(null));
+            Assert.Throws<ArgumentNullException>(() => SlhDsa.ImportSlhDsaPublicKey(null, ReadOnlySpan<byte>.Empty));
+            Assert.Throws<ArgumentNullException>(() => SlhDsa.ImportSlhDsaSecretKey(null, ReadOnlySpan<byte>.Empty));
+            Assert.Throws<ArgumentNullException>(() => SlhDsa.ImportSlhDsaPrivateSeed(null, ReadOnlySpan<byte>.Empty));
+        }
+
+        [Theory]
+        [MemberData(nameof(AlgorithmsData))]
+        public static void ArgumentValidation(SlhDsaAlgorithm algorithm)
+        {
+            using SlhDsa slhDsa = SlhDsa.GenerateKey(algorithm);
+
+            Assert.Throws<ArgumentNullException>(() => slhDsa.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>.Empty, null));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.ExportEncryptedPkcs8PrivateKeyPem(ReadOnlySpan<byte>.Empty, null));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.ExportEncryptedPkcs8PrivateKeyPem(ReadOnlySpan<char>.Empty, null));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>.Empty, null, Span<byte>.Empty, out _));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>.Empty, null, Span<byte>.Empty, out _));
```

Comment:
These seem more like either the "testing the template method pattern" tests or a "all the kinds of instance" test.  But they're not quite the same as the static import tests below, so should probably be separated.